portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,26 +33,43 @@
 /** Servlet that returns comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  
-  // Create ArrayList for comments
-  List<String> comments = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Convert ArrayList of comments to JSON using Gson
-    String json = new Gson().toJson(comments);
+    Query query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String newComment = (String) entity.getProperty("text");
+      long time = (long) entity.getProperty("time");
+
+      Comment comment = new Comment(id, newComment, time); 
+      comments.add(comment);
+    }
+
+    Gson gson = new Gson();
+    String json = gson.toJson(comments);
 
-    // Send the JSON as the response
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
 
+  @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form.
-    String userMessage = request.getParameter("comment-text");
+    String userInput = request.getParameter("text");
+    long time = System.currentTimeMillis();
 
-    // Add user input message to the list of comments.
-    comments.add(userMessage);
+    // Store comment in the comment entity.
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("text", userInput);
+    commentEntity.setProperty("time", time); 
```

Comment:
(Optional): Just as an FYI (since there's not a lot of properties) you could write this as:
```suggestion
    Entity commentEntity = Entity.newBuilder("Comment")
        .set("text", userInput)
        .set("time", time)
        .build();
```

Reply:
Thank you for the suggestion! For this and the previous suggestion, I somehow received the following error. Is there something that I need to import? Thank you!

`[ERROR] /home/tshafikah/software-product-sprint/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:[72,34] cannot find symbol
[ERROR]   symbol:   method newBuilder(java.lang.String)
[ERROR]   location: class com.google.appengine.api.datastore.Entity`

Reply:
Oh my, it turns out I was looking at the wrong Datastore library :facepalm: